mitiq/calibration/calibrator.py

```diff
@@ -178,12 +180,49 @@ def best_strategy(self, results: List[Dict[str, Any]]) -> Strategy:
             parameters that performed best.
         """
         best_improvement_factor = 1.0
+        num_circuits = len(self.settings.circuit_types)
+
         strategy = DefaultStrategy
-        for circuit in results:
-            for result in circuit["mitigated_values"]["ZNE"]["results"]:
-                if result["improvement_factor"] > best_improvement_factor:
-                    best_improvement_factor = result["improvement_factor"]
-                    strategy = result["strategy"]
+
+        def filter_on_strategy(
```

Comment:
Docstring is missing

Reply:
Wasn't sure if one was needed here given the simplicity of the function, but since it came up I'll add one.